### PR TITLE
feat(replays): display error count by project

### LINKED ISSUE #48363: Replay Details > Error count in the top-right

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary
This change adds the error count grouped by project. 
~Note this depends on: https://github.com/getsentry/sentry/pull/48339 - for legibility ive cut this PR against that branch instead of master.~

**No errs:**
image

**1 project w/ errs:**
image

**2 project w/ errs:**
image

**3+ project w/ errs:**
image

Relates to: https://github.com/getsentry/sentry/issues/48249
Closes: https://github.com/getsentry/sentry/issues/48363

### CODE REVIEW

**Comment by eliashussary** [comment]
Note to self: handle many projects in the UI. ideal state, we stack them and count total.. perhaps fan out on hover etc etc.

For now show 2 projects + total count of remaining

File: static/app/views/replays/detail/replayMetaData.tsx [hunk 1]
```diff
@@ -29,6 +33,17 @@ function ReplayMetaData({replayRecord}: Props) {
     },
   };
 
+  const errorCountByProject = useMemo(
```
Comment (ryan953) [inline static/app/views/replays/detail/replayMetaData.tsx:36]: we need `useMemo` when there's only expected to be like single-digits number in the aray?
the header can't really scale more to render more than that.

**Review by ryan953 — APPROVED** [review]

File: static/app/views/replays/detail/replayMetaData.tsx [hunk 2]
```diff
@@ -72,6 +103,48 @@ function ReplayMetaData({replayRecord}: Props) {
   );
 }
 
+function StackedErrorCount({
+  errorCounts,
+}: {
+  errorCounts: Array<{count: number; project: Project | undefined}>;
```
Comment (ryan953) [inline static/app/views/replays/detail/replayMetaData.tsx:109]: `undefined` is in here?

**Review by ryan953 — APPROVED** [review]

File: static/app/views/replays/detail/replayMetaData.tsx (hunk 2 above)
Comment (eliashussary) [reply static/app/views/replays/detail/replayMetaData.tsx:109]: return of `arr.find(predicate)` always returns `T | undefined`. it's gonna be extremely unlikely that it occurs, but at least the typing is there and we handle this case by rendering null

File: static/app/views/replays/detail/replayMetaData.tsx (hunk 1 above)
Comment (eliashussary) [reply static/app/views/replays/detail/replayMetaData.tsx:39]: i don't quite understand; but if we're saying that `useMemo` here is an over optimization, yes it is lol

**Pull request merged**